EasyPost/CreditCard.cs

```diff
@@ -0,0 +1,124 @@
+        private static async Task<CreditCard> GetPaymentMethodByPriority(Priority priority)
+        {
+            PaymentMethod paymentMethods = await PaymentMethod.All();
+
+            CreditCard? paymentMethod = null;
+            switch (priority)
+            {
+                case Priority.Primary:
+                    paymentMethod = paymentMethods.primary_payment_method;
+                    break;
+                case Priority.Secondary:
+                    paymentMethod = paymentMethods.secondary_payment_method;
+                    break;
+                default:
+                    break;
+            }
+
+            if (paymentMethod == null)
+            {
+                throw new Exception("The chosen payment method is not a valid method. Please try again.");
+            }
+
+            return paymentMethod;
+        }
+
+        /// <summary>
+        ///     Fund a credit card.
+        /// </summary>
+        /// <param name="amount">Amount to fund.</param>
+        /// <param name="paymentMethodId">ID of payment method to fund.</param>
+        /// <returns>An EasyPost.CreditCardFund instance.</returns>
+        public static async Task<CreditCardFund> Fund(string amount, string paymentMethodId)
+        {
+            Request request = new Request($"credit_cards/{paymentMethodId}/charge", Method.Post);
+            request.AddParameters(new Dictionary<string, object>
+            {
+                {
+                    "amount", amount
+                }
+            });
+
+            return await request.Execute<CreditCardFund>();
+        }
+
+        /// <summary>
+        ///     Fund a credit card.
+        /// </summary>
+        /// <param name="amount">Amount to fund.</param>
+        /// <param name="priority">Which type of payment method to fund.</param>
+        /// <returns>An EasyPost.CreditCardFund instance.</returns>
+        public static async Task<CreditCardFund> Fund(string amount, Priority priority)
+        {
+            CreditCard paymentMethod = await GetPaymentMethodByPriority(priority);
+
+            return await Fund(amount, paymentMethod.id);
+        }
+
+        /// <summary>
+        ///     Delete a credit card.
+        /// </summary>
+        /// <param name="paymentMethodId">ID of the payment method to delete.</param>
+        /// <returns>Whether the request was successful or not.</returns>
+        public static async Task<bool> Delete(string paymentMethodId)
+        {
+            Request request = new Request($"credit_cards/{paymentMethodId}", Method.Delete);
+
+            return await request.Execute();
+        }
+
+        /// <summary>
+        ///     Delete a credit card.
+        /// </summary>
+        /// <param name="priority">Which payment method to delete.</param>
+        /// <returns>Whether the request was successful or not.</returns>
+        public static async Task<bool> Delete(Priority priority)
```

Comment:
Why do we support deleting a credit card by its priority? This wasn't in the documentation

Reply:
Yeah, I added it thinking it would be a helpful "shortcut".
@Justintime50 care to weigh in?

Reply:
Although I like the idea, the other 6 libs are already done and not built this way. For now, let’s do it by ID. This will also keep it consistent with how all our other functions for everything work (provide an ID). I’m a fan of revisiting potentially though. Worst case scenario is someone using this simply retrieves their payment methods and makes a second call to delete doing so programmatically.